resources/google_agent.rb

```diff
@@ -0,0 +1,177 @@
+# Copyright:: 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# To learn more about Custom Resources, see https://docs.chef.io/custom_resources/
+# Use upgrade because it'll fall back to install if software isn't installed already
+# By using upgrade as the default action, if the admin changes the version attribute,
+# then the software will update. If this was set to just install, changing the version
+# attribute would not result in servers being upgraded.
+default_action :upgrade
+resource_name :google_agent
```

Comment:
For branding consistency, can we make this `google_cloud_ops_agents`? Same with the file name.

Reply:
Yeah, we can sync about this in Google Hangout, or the meeting tomorrow.